### PR TITLE
(BKR-1189) Fix port mapping

### PR DESCRIPTION
**Pull request by rishijavia**
The port mapping is for port 22 that is used to ssh. This maps a static random 4 digit port when the container is created so a random port is not assigned. 

When a random port is assigned it would change it when the container is restarted and beaker would still be using old port and would not be able to ssh.

### CODE REVIEW

**Comment by Ananasr** [comment]
Can you make a new release asap please ? I really need these latest PRs !

**Comment by rishijavia** [comment]
I'll try to get this merged and release cut off a new version today. Sorry for the delay

File: lib/beaker/hypervisor/docker.rb [hunk 1]
```diff
@@ -102,7 +113,7 @@ def provision
         fix_ssh(container) if @options[:provision] == false
 
         @logger.debug("Starting container #{container.id}")
-        container.start({"PublishAllPorts" => true, "Privileged" => true})
```
Comment (tvpartytonight) [inline lib/beaker/hypervisor/docker.rb]: What is the a reason you moved these parameters to `container_opts`? Is there a change in behavior by doing so, or is it just a refactor?

Comment (rishijavia) [inline lib/beaker/hypervisor/docker.rb]: If any options are passed with `container.start` then those are the only options that are used. So originally any options provided by `container_opts` were not being used at all and therefore consolidating all the options in `container_opts`

Comment (tvpartytonight) [inline lib/beaker/hypervisor/docker.rb]: `container_opts` is the parameter used in `::Docker::Container.create(container_opts)` on [line 104](https://github.com/rishijavia/beaker-docker/blob/047407f64220862e8c7be9c2e728ee2b77c39e90/lib/beaker/hypervisor/docker.rb#L104) of this PR, so it is definitely still being used. I'm just wondering if the options parameter provided there are read in the same way as they are when provide to `start`.

Comment (tvpartytonight) [inline lib/beaker/hypervisor/docker.rb]: I'm still a bit unclear why removing the parameters has this effect; the `start` method only makes an api call with parameters if they are provided, and doesn't seem to do any sort of reading of currently set parameters from the logic seen [here](https://github.com/swipely/docker-api/blob/master/lib/docker/container.rb#L201-L204).

**Comment by tvpartytonight** [comment]
I'm fine merging this, even though I still have some unanswered questions about what is happening inside the docker-api gem. Thanks to @rishijavia's work, we can just do a new release if there are any issues.

File: lib/beaker/hypervisor/docker.rb (hunk 1 above)
Comment (rishijavia) [inline lib/beaker/hypervisor/docker.rb]: looking at their [API](https://docs.docker.com/engine/api/v1.30/#operation/ContainerStart) they dont have any options that are provided with `start` (i understand that the code still support taking options on `start`). but looks like [creating](https://docs.docker.com/engine/api/v1.30/#operation/ContainerCreate) container is where the options are passed

i also tried a combination of keeping the options provided with `start` as is and passing other options with `create` but all the options passed with `create` is being overwritten 😞

File: lib/beaker/hypervisor/docker.rb [hunk 2]
```diff
@@ -70,6 +70,17 @@ def provision
         container_opts = {
           'Image' => image_name,
           'Hostname' => host.name,
+          'name' => host.name,
```
Comment (jovrum) [inline lib/beaker/hypervisor/docker.rb]: I think this introduced a regression. The SUT containers are now named the same as their hostname, whereas before they had randomly generated names. When running two instances of the same nodeset simultaneously, or two separate nodesets with overlapping hostnames, beaker fails with the following error message:
```
An error occurred while loading ./spec/acceptance/class_spec.rb.

Failure/Error: require 'beaker-rspec'

Docker::Error::ConflictError:

  Conflict. The container name "/centos-6-x64" is already in use by container "5d843a8281173c6d3048432301fffc625301816eeba9c945822525a8a195001f". You have to remove (or rename) that container to be able to reuse that name.
```

**Pull request merged**